src/celengine/astroobj.h

```diff
@@ -21,7 +21,15 @@ class AstroObject
     AstroCatalog::IndexNumber getIndex() const { return m_mainIndexNumber; }
     void setIndex(AstroCatalog::IndexNumber);
 
+    static constexpr AstroCatalog::IndexNumber MaxAutoIndex = UINT32_MAX - 1;
+    static AstroCatalog::IndexNumber autoIndex(bool decrement = true);
```

Comment:
instead of adding function with flag, introduce 2 functions:
`getNextIndex` and `getCurrentIndex`

they should be `inline`

Reply:
maybe getNextIndex is not a correct name, maybe better getCurrentIndexAndUpdate